public/src/main/java/com/rackspace/salus/telemetry/api/services/TestMonitorAndEventTaskService.java

```diff
@@ -39,56 +41,82 @@
-                    .setMetrics(metrics);
-            testTaskResult = eventTaskApi
-                    .performTestTask(tenantId, testTaskRequest);
-            if (!CollectionUtils.isEmpty(testTaskResult.getErrors()) || testTaskResult.getData() == null) {
-                return new TestMonitorAndEventTaskResponse(
-                        new TestMonitorAndEventTask(testMonitorResult, testTaskResult),
-                        CollectionUtils.isEmpty(testTaskResult.getErrors()) ? List.of("Unable to test-events") : testTaskResult.getErrors());
-            }
+      testTaskResult = eventTaskApi
+          .performTestTask(tenantId, testTaskRequest);
 
-            return new TestMonitorAndEventTaskResponse(
-                    new TestMonitorAndEventTask(testMonitorResult, testTaskResult), null);
-        } catch (RemoteServiceCallException e) {
-            return new TestMonitorAndEventTaskResponse(
-                    new TestMonitorAndEventTask(testMonitorResult, testTaskResult), List.of(String
-                    .format("An unexpected internal error occurred: %s", e.getMessage())));
-        }
+      if (testTaskResult == null || !CollectionUtils.isEmpty(testTaskResult.getErrors())
+          || testTaskResult.getData() == null) {
+        return new TestMonitorAndEventTaskResponse().setData(new TestMonitorAndEventTask()
+            .setMonitor(
+                new TestMonitorResultData().setMetrics(testMonitorResult.getData().getMetrics())))
+            .setErrors(CollectionUtils.isEmpty(testMonitorResult.getErrors()) ? List
+                .of("Unable to get test event data") : testMonitorResult.getErrors());
+      }
 
+      return new TestMonitorAndEventTaskResponse().setData(new TestMonitorAndEventTask()
+          .setMonitor(
+              new TestMonitorResultData().setMetrics(testMonitorResult.getData().getMetrics()))
+          .setTask(new TestTaskResultResultData().setEvents(testTaskResult.getData().getEvents())
+              .setStats(testTaskResult.getData()
+                  .getStats())));
+    } catch (RemoteServiceCallException e) {
+      TestMonitorAndEventTaskResponse testMonitorAndEventTaskResponse = new TestMonitorAndEventTaskResponse();
+      if (testMonitorResult.getData() != null) {
+        testMonitorAndEventTaskResponse.setData(
+            new TestMonitorAndEventTask().setMonitor(
+                new TestMonitorResultData().setMetrics(testMonitorResult.getData().getMetrics())));
+      } else if (testTaskResult.getData() != null) {
+        new TestMonitorAndEventTask().setTask(
+            new TestTaskResultResultData().setEvents(testTaskResult.getData().getEvents())
+                .setStats(testTaskResult.getData()
+                    .getStats()));
+      } else {
+        return new TestMonitorAndEventTaskResponse().setErrors(List.of(String
+            .format("An unexpected internal error occurred: %s", e.getMessage())));
+      }
     }
+    return null;
```

Comment:
This `return null` is masking an issue. Should be
```suggestion
      return testMonitorAndEventTaskResponse;
    }
```